### PR TITLE
Fix #2203: Improve recent upload limit setting

### LINKED ISSUE #2203: Upload limit amount doesn't update EditTextPreference if hit cap
**Summary**

The upload limit amount can be edited to be over 500. While it doesn't break anything, it is weird that the SettingsFragment remembers this when you edit the setting again without exiting settings.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by domdomegg**
**Description**

Fixes #2203 

What changes did you make and why?
- Set editText of preference to 500 if set over the limit (see GIF)
- Use previous setting rather than 100 if invalid
- Print a stack trace if invalid number entered, as shouldn't be possible
- Removed duplicated code
- Remove unused field "REQUEST_CODE_WRITE_EXTERNAL_STORAGE"
- Removed unused imports in SettingsFragment
- In `preferences.xml` set `android:inputType` of the pref to be `number`, stopping users from entering decimal points (which could cause `NumberFormatException`s when parsed as an integer)
- In `strings.xml` removed unused related string `number_of_uploads`
- In `strings.xml` changed preference title from "Set Recent Upload Limit" to just "Recent upload limit" to standardise capitalisation among settings and remove redundant "Set"

**Tests performed**

Tested `2.9.0-debug-fix-2203~98ec9f686` on `Galaxy Nexus (emulator)` at API level `28`

**GIF**

| Before | After |
| - | - |
| before | after |

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java [hunk 1]
```diff
@@ -68,40 +61,42 @@ public void onCreate(Bundle savedInstanceState) {
...
+            authorName.setEnabled((Boolean) newValue);
             return true;
         });
 
         final EditTextPreference uploadLimit = (EditTextPreference) findPreference("uploads");
-        int uploads = prefs.getInt(Prefs.UPLOADS_SHOWING, 100);
-        uploadLimit.setText(uploads + "");
-        uploadLimit.setSummary(uploads + "");
+        int currentUploadLimit = prefs.getInt(Prefs.UPLOADS_SHOWING, 100);
+        uploadLimit.setText(Integer.toString(currentUploadLimit));
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java]: Some explanation comment about  "100" would be nice.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -190,10 +190,9 @@
   <string name="map_theme_dark">mapbox://styles/mapbox/traffic-night-v2</string>
   <string name="mapbox_commons_app_token">pk.eyJ1IjoibWFza2FyYXZpdmVrIiwiYSI6ImNqMmxvdzFjMTAwMHYzM283ZWM3eW5tcDAifQ.ib5SZ9EVjwJe6GSKve0bcg</string>
   <string name="_2fa_code">2FA Code</string>
-  <string name="number_of_uploads">My Recent Upload Limit</string>
   <string name="maximum_limit">Maximum Limit</string>
   <string name="maximum_limit_alert">Unable to display more than 500</string>
-  <string name="set_limit">Set Recent Upload Limit</string>
+  <string name="set_limit">Recent upload limit</string>
```
Comment (neslihanturan) [inline app/src/main/res/values/strings.xml]: I think "Set recent upload limit" title would be better. I agree about unnecessary capitals but I do think we should follow imperative form in preferences. See:
- Use external storage
- Display nearby notification

File: app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java (hunk 1 above)
Comment (domdomegg) [reply app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java]: Agreed, will fix. 100 is default/fallback setting.

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (domdomegg) [reply app/src/main/res/values/strings.xml]: I disagree with imperative form in preferences. I think the two examples you have should keep their verbs as you are toggling wether that behaviour happens. However, "set" is what you're doing right now.

I.e. clicking "display nearby notification" doesn't display the notification but sets whether the app should display the nearby notification. Likewise all the settings have an implied "set" or "set whether the app".

For comparison in android sound settings on my phone there is:
- Phone ringtone (parallels with setting a custom value like upload count)
- Vibrate for calls (parallels with setting whether the app should perform an action like display nearby notification)

Also night mode, default license are currently not.

Comment (neslihanturan) [reply app/src/main/res/values/strings.xml]: Hmm, from this point of view, you are right. But do you think user will notice this setting is something they can click and change if we say "Recent upload limit" ?

Comment (domdomegg) [reply app/src/main/res/values/strings.xml]: I believe so - everything else in settings is something they can click. This design pattern is used in android's own settings and is already true of "default license".

Comment (domdomegg) [reply app/src/main/res/values/strings.xml]: Just did a bit more research, Android's documentation states titles should avoid "Generic terms, such as **set**, change, edit, modify, manage, use, select, or choose."

Also you can see none of the screenshots there (which I believe are from the Pixel settings app) have "set", some with options that are similar (as in they are just modifying some integer value)

https://source.android.com/devices/tech/settings/settings-guidelines

File: app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java (hunk 1 above)
Comment (domdomegg) [reply app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java]: I've fixed this, and also the arbitrary "500" limit too.

**Comment by domdomegg** [comment]
Closing as @vanshikaarora did this again in #2292, and resolving the merge conflicts will be too painful.

**Pull request closed without merging**